debian/postinst

```diff
@@ -72,6 +75,34 @@ configure)
 		echo "ERROR: postgres id $(id -u postgres) != 65437" >&2
 		exit 1
 	fi
+
+	# We are switching to MAC address based network interface names only in AWS environment
+	# from 6.0.10.0 onwards. To handle this change, we leverage the default 80-net-setup-link.rules udev
+	# rules file. We make a copy of this file under /etc and then change the naming policy to MAC address
+	# based names. Please see DLPX-75209 for more context on this change.
+	# Though unlikely, it is still remotely possible that the /usr/lib/udev/rules.d/80-net-setup-link.rules
+	# file either gets removed from the base OS distribution or gets replaced with another file to serve
+	# the exact/similar purpose. We must detect and inspect such a change and update the renaming logic if
+	# needed. Ideally, we should also detect changes in the contents of this file but we doubt there is an
+	# obvious way of accomplishing this today.
+	if [[ ! -e "$DEFAULT_UDEV_NET_SETUP_LINK_DIR"/80-net-setup-link.rules ]]; then
+		echo "$DEFAULT_UDEV_NET_SETUP_LINK_DIR/80-net-setup-link.rules found missing, investigation needed to understand " \
+			"if its location has changed or if the file itself has been replaced with another udev rules file" >&2
+		exit 1
+	fi
+	if [[ "$(get-appliance-platform)" == "aws" ]]; then
+		# If there are any changes in the specific line of the rules file that we are modifying, we fail early
+		# to allow us the opportunity to investigate the changes and update our own changes accordingly.
+		grep "^NAME==\"\", ENV{ID_NET_NAME}\!=\"\", NAME=\"\$env{ID_NET_NAME}\"" \
```

Comment:
instead of using `grep`.. could we instead modify the `sed` command to be more specific about the line we're expecting? I don't feel strongly, so I'm fine as-is.. but just curious if we could simplify the logic and remove `grep`, but still maintain the "fail if the line isn't what we expect" semantics..